docs/examples/aws_ebs_volume_idempotence.rb

```diff
@@ -0,0 +1,34 @@
+require 'chef/provisioning/aws_driver'
+
+with_driver 'aws::us-west-2'
+
+key = aws_key_pair 'ref-key-pair-ebs'
+key.run_action(:create)
+
+ref_machine1 = machine 'ref-machine-1'
+ref_machine1.run_action(:converge)
+
+ebs_volume = aws_ebs_volume 'ref-volume-ebs' do
+  action :nothing
+  machine ref_machine1.name
+  device '/dev/xvdf'
+  availability_zone 'us-west-2a'
+  size 1
+end
+
+ebs_volume.run_action(:create)
+ebs_volume.run_action(:create) # up to date
+ebs_volume.run_action(:attach)
+ebs_volume.run_action(:attach) # up to date
+ebs_volume.run_action(:detach)
+ebs_volume.run_action(:detach) # up to date
+ebs_volume.run_action(:delete)
+ebs_volume.run_action(:delete) # up to date
```

Comment:
Just talked with @jkeiser - we're going to disable resource cloning for all the aws_* resources (https://github.com/chef/chef-provisioning-aws/blob/jk/awesome_security_groups/lib/chef/provisioning/aws_driver/super_lwrp.rb#L37-L40).  So you should be able to update this to look like:

```ruby
aws_key_pair 'ref-key-pair-ebs'

machine 'ref-machine-1' do
  machine_options bootstrap_options: { key_name: 'ref-key-pair-ebs' }
end

aws_ebs_volume 'ref-volume-ebs' do
  action :nothing
  machine ref_machine1.name
  device '/dev/xvdf'
  availability_zone 'us-west-2a'
  size 1
end

aws_ebs_volume 'ref-volume-ebs' do
  action :create
end
# up to date
aws_ebs_volume 'ref-volume-ebs' do
  action :create
end
aws_ebs_volume 'ref-volume-ebs' do
  action :attach
end
# up to date
aws_ebs_volume 'ref-volume-ebs' do
  action :attach
end
aws_ebs_volume 'ref-volume-ebs' do
  action :detach
end
# up to date
aws_ebs_volume 'ref-volume-ebs' do
  action :detach
end
aws_ebs_volume 'ref-volume-ebs' do
  action :delete
end
# up to date
aws_ebs_volume 'ref-volume-ebs' do
  action :delete
end
```

This will also prevent these actions from running at compile time.